### PR TITLE
Raise more errors and clarify what is going on

### PR DESCRIPTION
**Pull request by patrick-laa**
## Description of change
There are some errors we haven't been catching but should
And the logic was a little hard to follow

### CODE REVIEW

File: app/jobs/notify_subscriber.rb [hunk 1]
```diff
@@ -13,17 +13,24 @@ def self.perform_later(subscriber_id, submission)
   def perform(subscriber_id, submission)
     raise_error = false
     subscriber = Subscriber.find(subscriber_id)
-    raise_error = handle_failure(subscriber) unless send_message_to_webhook(subscriber.webhook_url, submission)
+    unless send_message_to_webhook(subscriber.webhook_url, submission)
+      raise_error = handle_failure_and_decide_whether_to_raise_error(subscriber)
```
Comment (elken) [inline app/jobs/notify_subscriber.rb:17]: `handle_failure_maybe_raise`?

File: app/jobs/notify_subscriber.rb [hunk 2]
```diff
@@ -37,17 +44,20 @@ def send_message_to_webhook(webhook_url, submission)
...
   end
 
+  # Returns a boolean indicating whether the subscriber has been deleted.
   def delete_subscriber(subscriber)
     deletion_threshold = ENV.fetch("SUBSCRIBER_FAILED_ATTEMPT_DELETION_THRESHOLD", nil)
 
-    return unless deletion_threshold&.to_i&.positive? && deletion_threshold.to_i <= subscriber.failed_attempts
+    return false unless deletion_threshold&.to_i&.positive? && deletion_threshold.to_i <= subscriber.failed_attempts
 
     subscriber.destroy!
```
Comment (elken) [inline app/jobs/notify_subscriber.rb:59]: Why doesn't `destroy!` return said bool?

**Review by elken — APPROVED** [review]

File: app/jobs/notify_subscriber.rb (hunk 2 above)
Comment (patrick-laa) [reply app/jobs/notify_subscriber.rb:59]: Ah, because destroy either returns a `Subscriber` object or raises an exception. Obviously a `Subscriber` is truthy, but for the sake of clarity I'm trying to use booleans `true/false` rather than `object/nil`

File: app/jobs/notify_subscriber.rb (hunk 1 above)
Comment (patrick-laa) [reply app/jobs/notify_subscriber.rb:17]: Ooh that's nicer.

Comment (elken) [reply app/jobs/notify_subscriber.rb:17]: Naming things is hard, I do kinda wish Ruby had some kind of standard documentation thing like JSDoc
